### DISCUSSION TITLE
Missing builders in Version 6

### DISCUSSION THREAD

**Discussion by jcrada** [danvick/flutter_form_builder#784] [General]
Hi Dan. Love your work!

On version 6 you removed several builders (eg searchable dropdown). I understand it is to bring project to compatibility with latest flutter nullsafety.

Will the removed builders be back once dependencies are upgraded or replaced?

Many thanks and keep doing such excellent work!

**Comment by danvick** [comment]
Hi @jcrada,

I'm glad you find the package useful.

Null-safety did expose a major flaw of this package in that we have been dependent on so many external packages, some of which aren't being actively or well maintained.
In the past, we've had to submit PRs to other repos after issues are raised in this repo about issues emanating from the dependencies; but with null-safety, this wasn't remotely possible.

This made us rethink the package structure. We decided to remove all fields with external dependencies from the package in version 6.

Not to worry though, the plan is to split up the package into smaller packages but we'll end up with all fields we had initially - as long as the dependencies are properly maintained.
As from ~7.0.0, we'll have these packages:

[flutter_form_builder](https://pub.dev/packages/flutter_form_builder): the core packages with fundamental building blocks and common fields that are native to Flutter
[form_builder_validators](https://pub.dev/packages/form_builder_validators) - for validation (Can be used separately without FormBuilder)
[form_builder_extra_fields](https://pub.dev/packages/form_builder_extra_fields) - additional fields that depend on external packages (such as your beloved FormBuilderSearchableDropdown 😀)
